File: Programming-Basics-With-JavaScript/Exams/01-Old-Exams/04-PB-Exam-July-2020/05.bestPlayer.js
function bestPlayer(input) {

    let index = 0;
    let current = input[index++];

    let bestPlayer = "";
    let bestscore = 0;

    while (current !== "END") {
        let playerName = current;
        let score = Number(input[index++]);
        if (score > bestscore) {
            bestscore = score;
            bestPlayer = playerName;
            
            // Ако футболистът е вкарал 10 или повече гола, програмата трябва да спре.
            if (score > 9) {
                break;
            }
        }

        current = input[index++];
    }

    console.log(`${bestPlayer} is the best player!`);
    if (bestscore >= 3) {
        console.log(`He has scored ${bestscore} goals and made a hat-trick !!!`);
    } else {
        console.log(`He has scored ${bestscore} goals.`);
    }
}
bestPlayer(["Neymar"
    , "2"
    , "Ronaldo"
    , "1"
    , "Messi"
    , "3"
    , "END"]);

File: Python_Basics/Exams/08.python_basics_pre-exam_10_Jun_2023/05.best_player.py
player_name = input()

# Хет-трик е, когато футболистът е вкарал 3 или повече гола.
is_made_a_hat_trick = False
best_player = ''
goals = 0

while player_name != 'END':
    goals_count = int(input())

    if goals_count > goals:
        best_player = player_name
        goals = goals_count
        if goals_count >= 3:
            is_made_a_hat_trick = True

    # Ако футболист е вкарал 10 или повече гола, програмата трябва да спре.
    if goals_count >= 10:
        break

    player_name = input()

print(f'{best_player} is the best player!')

if is_made_a_hat_trick:
    print(f'He has scored {goals} goals and made a hat-trick !!!')
else:
    print(f'He has scored {goals} goals.')
